### PR TITLE
CBAPI-2034 Defense Migration Guide

### PR DESCRIPTION
**Pull request by llyon-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

https://jira.carbonblack.local/browse/CBAPI-2034

## Pull Request Description

Add Platform Search info to the migration guide
Overhaul the migration guide structure for readability

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Build the documentation locally using the [instructions in the project REAMDE](https://github.com/carbonblack/carbon-black-cloud-sdk-python#building-the-documentation)

### CODE REVIEW

**Comment by llyon-cb** [comment]
CBAPI-2034 lists the following:

"Migration information provided for customers to move from

CBAPI Defense Events to CBC SDK Enriched Events
CBAPI Defense Processes to CBC SDK Processes Search
CBAPI Defense Alerts to CBC SDK Alerts
CBAPI Defense Devices to CBC SDK Devices"

However, we don't have Defense Processes or Alerts in CBAPI. I have included information in the "Replaced Modules" section of the migration guide about the changes to Platform search that I think are sufficient

**Comment by llyon-cb** [comment]
If you have trouble with your build environment, here's the compiled HTML doc for my changes:

[porting-guide.html.zip](https://github.com/carbonblack/carbon-black-cloud-sdk-python/files/5692333/porting-guide.html.zip)

File: docs/porting-guide.rst [hunk 1]
```diff
@@ -1,49 +1,311 @@
...
++----------------------------+-------------------------------+
+| CBAPI Name (old)           | CBC SDK Name (new)            |
++============================+===============================+
+| ``cbapi.psc.defense``      | ``cbc_sdk.endpoint_standard`` |
++----------------------------+-------------------------------+
+| ``cbapi.psc.livequery``    | ``cbc_sdk.audit_remediation`` |
++----------------------------+-------------------------------+
+| ``cbapi.psc.threathunter`` | ``cbc_sdk.enterprise_edr``    |
++----------------------------+-------------------------------+
+| Does not exist in CBAPI    | ``cbc_sdk.platform``          |
```
Comment (avanbrunt-cb) [inline docs/porting-guide.rst:39]: We did have the new device model in `cbapi.psc`

File: docs/porting-guide.rst [hunk 2]
```diff
@@ -1,49 +1,311 @@
...
+Some modules have been moved to a more appropriate location.
+
++-----------------------------+------------------------------+
+| CBAPI Name (old)            | CBC SDK Name (new)           |
++=============================+==============================+
+| ``cbapi.example_helpers``   | ``cbc_sdk.helpers``          |
++-----------------------------+------------------------------+
+| ``cbapi.psc.alerts_query``  | ``cbc_sdk.platform.alerts``  |
++-----------------------------+------------------------------+
+| ``cbapi.psc.devices_query`` | ``cbc_sdk.platform.devices`` |
```
Comment (avanbrunt-cb) [inline docs/porting-guide.rst:85]: Do we want to mention this format or should we be pushing them to be more consistent and use

```
from cbc_sdk.platform import Alerts, Devices
```

File: docs/porting-guide.rst (hunk 1 above)
Comment (llyon-cb) [reply docs/porting-guide.rst]: I've updated this

File: docs/porting-guide.rst (hunk 2 above)
Comment (llyon-cb) [reply docs/porting-guide.rst]: I think the more consistent format `cbc_sdk.platform` is best, I've updated this

**Review by avanbrunt-cb — APPROVED** [review]

**Review by abowersox-cb — APPROVED** [review]
You're a lot less terse than I am. :)

**Pull request merged**